# Análisis de Datos de Seguros - Enfoque Step-by-Step y Caso de Negocio

Este notebook está diseñado para que resuelvas problemas paso a paso utilizando Python (usando funciones básicas como .mean(), filtros, value_counts y groupby) y que puedas comparar los resultados con el objetivo propuesto. Además, se plantea un caso de negocio ficticio en el que se asigna un gasto en marketing (CAC) para evaluar la rentabilidad y tomar decisiones estratégicas.

El objetivo es evaluar tu habilidad para:
- Explorar y analizar datos en Python.
- Resolver problemas de análisis paso a paso.
- Comparar resultados con objetivos estratégicos.
- Formular conclusiones y recomendaciones basadas en los datos.

¡Comencemos!

In [1]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
!unzip WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip

--2025-03-13 01:08:30--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip [following]
--2025-03-13 01:08:30--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 352796 (345K) [

## Paso 1: Exploración Inicial del Dataset
1. Carga el dataset `WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv`.
2. Revisa las primeras filas, dimensiones y la información general del dataset.

_Observa la estructura de los datos: ¿Qué columnas tienes? ¿Se nota algún dato atípico?_

In [2]:
# Cargar el dataset y ver primeras filas
import pandas as pd
import numpy as np

df = pd.read_csv('WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv')
pd.set_option('display.max_columns', None)


## Paso 2: Análisis del Customer Lifetime Value (CLV)
1. Calcula el CLV promedio y su desviación estándar.
2. Reflexiona: ¿Qué indica una alta desviación? ¿Podrían existir segmentos de clientes con comportamientos muy distintos?

In [3]:
clv_mean = df["Customer Lifetime Value"].mean()
clv_std = df["Customer Lifetime Value"].std()
print(f"El CLV promedio es {clv_mean} y la desviación estándar es {clv_std}")

El CLV promedio es 8004.940474987081 y la desviación estándar es 6870.967608356924


**¿Qué indica una alta desviación? **
rta:Una alta desviación estándar indica que existen diferencias significativas en el valor que los clientes aportan
**¿Podrían existir segmentos de clientes con comportamientos muy distintos?**
rta:Los clientes cuentan con comportamientos muy distintos, puesto que la deviación es significativa.

## Paso 3: Evaluación de la Columna Response
1. Usa `value_counts()` para ver cuántos clientes respondieron "Yes" y "No".
2. Calcula el porcentaje de respuestas "Yes".

Pregunta: ¿La proporción de "Yes" es baja? ¿Qué podría indicar esto respecto a la estrategia de respuesta?

In [ ]:
df["Response"].value_counts()

Response
No     7826
Yes    1308
Name: count, dtype: int64

In [ ]:
response_counts = df["Response"].value_counts()
(response_counts.get('Yes', 0) / response_counts.sum()) * 100

14.320122618786948

**¿La proporción de "Yes" es baja? ¿Qué podría indicar esto respecto a la estrategia de respuesta?**
rta: si, la proporción de "yes" es bbaja puesto que solo es un 14% a comparación de las respuestas totales, por lo tanto se deben crear propuestas mas atractivas para el cliente o mejorar los canales de comunicación.


## Paso 4: Análisis de la Prima Mensual (Monthly Premium Auto) por Coverage
1. Agrupa los datos por `Coverage` y calcula la media de `Monthly Premium Auto`.
2. Compara los promedios obtenidos para cada tipo de Coverage.

Pregunta: ¿Existen diferencias notables entre las coberturas? ¿Qué implicaciones tendría esto en la estrategia de precios?

In [ ]:
df.groupby("Coverage")["Monthly Premium Auto"].mean()


Coverage
Basic        82.173851
Extended    103.579504
Premium     133.381068
Name: Monthly Premium Auto, dtype: float64

In [ ]:
Monthly_Premium_Auto = df.groupby("Coverage")["Monthly Premium Auto"].mean()
max_coverage = premium_by_coverage.idxmax()
min_coverage = premium_by_coverage.idxmin()
diff = premium_by_coverage.max() - premium_by_coverage.min()

print(f"\nLa cobertura con la prima mensual promedio más alta es: {max_coverage} "
      f"(${premium_by_coverage[max_coverage]:.2f})")
print(f"La cobertura con la prima mensual promedio más baja es: {min_coverage} "
      f"(${premium_by_coverage[min_coverage]:.2f})")
print(f"Diferencia entre la cobertura más cara y la más barata: ${diff:.2f}")



La cobertura con la prima mensual promedio más alta es: Premium ($133.38)
La cobertura con la prima mensual promedio más baja es: Basic ($82.17)
Diferencia entre la cobertura más cara y la más barata: $51.21


**¿Existen diferencias notables entre las coberturas?**
si exsiten diferencias notables entre las coberturas puesto que existe una cobertura más economica como lo es la Basic y una cobertuta con un precio mayor la cual es la premium.
 **¿Qué implicaciones tendría esto en la estrategia de precios?**
 estas diferencias de coberturas pueden impicar una segmentaciòn de clientes puesto que algunos estan dispuestos a pagar más que otros por el prooducto.

## Paso 5: Análisis del Ingreso (Income) según EmploymentStatus
1. Agrupa los datos por `EmploymentStatus` y calcula el ingreso promedio.

Pregunta: ¿Qué diferencias encuentras entre los grupos? ¿Cómo podría influir esto en la segmentación de clientes?

In [ ]:
df.groupby("EmploymentStatus")["Income"].mean()


EmploymentStatus
Disabled         20045.582716
Employed         56384.884521
Medical Leave    20292.770833
Retired          20554.960993
Unemployed           0.000000
Name: Income, dtype: float64

***¿Qué diferencias encuentras entre los grupos?***
se puede evidenciar que la mayor diferencia entre lo grupos se ve en el estado de empleo, puesto que dependiendo de ese se genera el ingreso quee reciben.

***¿Cómo podría influir esto en la segmentación de clientes?***
esto puede influir en la segmentaciòn de clientes en el ambito de estrategias de prescio, puesto que deppendiendiendo de su ingreso se pueden generar promociones o se pueden hacer ajustes para crear mayor competitividad.


## Paso 6: Análisis del CLV por Estado
1. Agrupa los datos por `State` y calcula el CLV promedio.

Pregunta: ¿Qué estado muestra el mayor CLV? ¿Qué factores regionales podrían explicar esto?

In [ ]:
df.groupby("State")["Customer Lifetime Value"].mean()

State
Arizona       7861.341489
California    8003.647758
Nevada        8056.706839
Oregon        8077.901191
Washington    8021.472273
Name: Customer Lifetime Value, dtype: float64

***¿Qué estado muestra el mayor CLV?***
 rta: el estado de Oregon.
***¿Qué factores regionales podrían explicar esto?***
 rta:los factores que pueden influenciar este factor puede ser la economía local, comportamiento del consumidor y políticas estatales del lugar.

## Paso 7: Análisis del Total Claim Amount según el Número de Pólizas
1. Separa a los clientes con más de 1 póliza de aquellos con 1 sola póliza.
2. Calcula el Total Claim Amount promedio para cada grupo.

Pregunta: ¿Qué diferencias encuentras y qué podrían indicar respecto al riesgo y comportamiento de reclamaciones?

In [ ]:
df[df["Number of Policies"] == 1]



Customer       State  Customer Lifetime Value Response  Coverage  \
0     BU79786  Washington              2763.519279       No     Basic   
4     HB64268  Washington              2813.692575       No     Basic   
16    FV94802      Nevada              2566.867823       No     Basic   
17    OE15005  California              3945.241604       No     Basic   
20    ZK25313      Oregon              2872.051273       No     Basic   
...       ...         ...                      ...      ...       ...   
9120  RN61682  California              2114.738469       No     Basic   
9121  BB10681  California              4140.648654       No  Extended   
9128  YM19146  California              4100.398533       No   Premium   
9130  PK87824  California              3096.511217      Yes  Extended   
9133  Y167826  California              2611.836866       No  Extended   

                 Education Effective To Date EmploymentStatus Gender  Income  \
0                 Bachelor           2/24/11         Employed      F   56274   
4                 Bachelor            2/3/11         Employed      M   43836   
16    High School or Below            2/6/11    Medical Leave      M   25049   
17                 College            1/5/11    Medical Leave      M   28855   
20    High School or Below           2/19/11         Employed      M   57749   
...                    ...               ...              ...    ...     ...   
9120              Bachelor           2/19/11       Unemployed      F       0   
9121               College           1/17/11       Unemployed      F       0   
9128               College            1/6/11         Employed      F   47761   
9130               College           2/12/11         Employed      F   21604   
9133               College           2/14/11       Unemployed      M       0   

     Location Code Marital Status  Monthly Premium Auto  \
0         Suburban        Married                    69   
4            Rural         Single                    73   
16        Suburban        Married                    67   
17        Suburban        Married                   101   
20        Suburban         Single                    74   
...            ...            ...                   ...   
9120      Suburban         Single                    72   
9121      Suburban       Divorced                   123   
9128      Suburban         Single                   104   
9130      Suburban       Divorced                    79   
9133      Suburban         Single                    77   

      Months Since Last Claim  Months Since Policy Inception  \
0                          32                              5   
4                          12                             44   
16                         14                              7   
17                         12                             59   
20                         31                             21   
...                       ...                            ...   
9120                       30                             69   
9121                        5                             35   
9128                       16                             58   
9130                       14                             28   
9133                        3                             90   

      Number of Open Complaints  Number of Policies     Policy Type  \
0                             0                   1  Corporate Auto   
4                             0                   1   Personal Auto   
16                            0                   1   Personal Auto   
17                            0                   1   Personal Auto   
20                            0                   1   Personal Auto   
...                         ...                 ...             ...   
9120                          5                   1   Personal Auto   
9121                          2                   1  Corporate Auto   
9128                          0                   

In [ ]:
df[df["Number of Policies"] > 1]

Customer       State  Customer Lifetime Value Response  Coverage  \
1     QZ44356     Arizona              6979.535903       No  Extended   
2     AI49188      Nevada             12887.431650       No   Premium   
3     WW63253  California              7645.861827       No     Basic   
5     OC83172      Oregon              8256.297800      Yes     Basic   
6     XZ87318      Oregon              5380.898636      Yes     Basic   
...       ...         ...                      ...      ...       ...   
9126  AC13887  California              4628.995325       No     Basic   
9127  TF56202  California              5032.165498       No     Basic   
9129  LA72316  California             23405.987980       No     Basic   
9131  TD14365  California              8163.890428       No  Extended   
9132  UP19263  California              7524.442436       No  Extended   

     Education Effective To Date EmploymentStatus Gender  Income  \
1     Bachelor           1/31/11       Unemployed      F       0   
2     Bachelor           2/19/11         Employed      F   48767   
3     Bachelor           1/20/11       Unemployed      M       0   
5     Bachelor           1/25/11         Employed      F   62902   
6      College           2/24/11         Employed      F   55350   
...        ...               ...              ...    ...     ...   
9126  Bachelor            1/9/11       Unemployed      M       0   
9127   College           2/12/11         Employed      M   66367   
9129  Bachelor           2/10/11         Employed      M   71941   
9131  Bachelor            2/6/11       Unemployed      M       0   
9132   College            2/3/11         Employed      M   21941   

     Location Code Marital Status  Monthly Premium Auto  \
1         Suburban         Single                    94   
2         Suburban        Married                   108   
3         Suburban        Married                   106   
5            Rural        Married                    69   
6         Suburban        Married                    67   
...            ...            ...                   ...   
9126      Suburban         Single                    67   
9127      Suburban       Divorced                    64   
9129         Urban        Married                    73   
9131      Suburban         Single                    85   
9132      Suburban        Married                    96   

      Months Since Last Claim  Months Since Policy Inception  \
1                          13                             42   
2                          18                             38   
3                          18                             65   
5                          14                             94   
6                           0                             13   
...                       ...                            ...   
9126                       25                             21   
9127                        6                             48   
9129                       18                             89   
9131                        9                             37   
9132                       34                              3   

      Number of Open Complaints  Number of Policies     Policy Type  \
1                             0                   8   Personal Auto   
2                             0                   2   Personal Auto   
3                             0                   7  Corporate Auto   
5                             0                   2   Personal Auto   
6                             0                   9  Corporate Auto   
...                         ...                 ...             ...   
9126                          0                   4  Corporate Auto   
9127                          0                   3   Personal Auto   
9129                          0                   2   Personal Auto   
9131                          3                   2  Corporate Auto   
9132                          0                   3 

In [ ]:
df[df["Number of Policies"] > 1]["Total Claim Amount"].mean()

430.6682882804692

In [ ]:
difference = multiple_policies - single_policy
print(f"La diferencia entre las dos poblaciones es: {difference}")

La diferencia entre las dos poblaciones es: -9.610243648168193


***¿Qué diferencias encuentras y qué podrían indicar respecto al riesgo y comportamiento de reclamaciones?***
La mayor diferencia que se ve es que los clientes con solo una poliza cuuentan con un pprecio mayor al que tienen más de una poliza, los distintos precios pueden implicar variaciones en las estrategias de suscripción y precios.

## Paso 8: Análisis del CLV según Vehicle Class
1. Agrupa los datos por `Vehicle Class` y calcula el CLV promedio para cada clase.

Pregunta: ¿Qué clase de vehículo se asocia a un mayor CLV y qué implicaciones podría tener esto en la estrategia de producto?

In [ ]:
df.groupby("Vehicle Class")["Customer Lifetime Value"].mean()

Vehicle Class
Four-Door Car     6631.726607
Luxury Car       17053.348399
Luxury SUV       17122.999134
SUV              10443.511816
Sports Car       10750.989331
Two-Door Car      6671.030732
Name: Customer Lifetime Value, dtype: float64

 **¿Qué clase de vehículo se asocia a un mayor CLV y qué implicaciones podría tener esto en la estrategia de producto?**
 El vehiculo que más se asocia a un mayor CLV es el luuxury SUV, esto podria tener implicaciones en la publicidad haciendo que se enfoque en segmentos más renables


## Paso 9: Análisis de Months Since Last Claim según Education
1. Agrupa los datos por `Education` y calcula el promedio de `Months Since Last Claim`.

Pregunta: ¿Existen diferencias notables entre los niveles educativos? ¿Qué podría indicar sobre la gestión de reclamos?

In [ ]:
 df.groupby("Education")["Months Since Last Claim"].mean()

Education
Bachelor                14.834789
College                 15.432301
Doctor                  15.260234
High School or Below    14.978642
Master                  15.199730
Name: Months Since Last Claim, dtype: float64

rta: no se evidencian diferencias notables entre los niveles educativos, esto puede indicar sobre la gestion de reclamos que existe un comportacmiento hoomogeneo en cuanto a los tiempo de educaciòn, además se puede hacer una estrategia aun más eficiente basandose en la anterior puesto que la diferencia no es grande.

## Paso 10: Análisis del Income según Marital Status y Gender
1. Realiza una segmentación cruzada agrupando por `Marital Status` y `Gender` para calcular el ingreso promedio.

Pregunta: ¿Qué diferencias observas? ¿Cómo podrían influir estas diferencias en la estrategia de segmentación de clientes?

In [ ]:
df.groupby(["Marital Status", "Gender"])["Income"].mean()

Marital Status  Gender
Divorced        F         37913.709450
                M         43681.934848
Married         F         44276.994962
                M         43304.110361
Single          F         23573.758120
                M         22005.776407
Name: Income, dtype: float64

Rta: las diferencias más notables son que las personas casadas tienen los ingresos más altos a compraciòn con las otras dos categorias, mientras que las parejas divorciadas tienen ingresos medios y las personas solteras son las personas con menos ingresos en comparaciòn de las distintas categorias.
Estas diferencias pueden influir en estrategias como lo seria ofertas enfocadas al estado civil, o campañas enfocadas al genero e ingreso de la persona haciendo que la oferta se vuelva màs atractiva.

## Paso 11: Distribución de la Columna Response
1. Usa `value_counts()` para mostrar cuántos clientes hay en cada grupo de Response.

Pregunta: ¿La distribución de respuestas sugiere que se debe ajustar la estrategia de comunicación?

In [ ]:
df["Response"].value_counts(normalize=True) * 100

Response
No     85.679877
Yes    14.320123
Name: proportion, dtype: float64

Rta: si se debe ajustar la estrategia de comunicación puesto que solo el 14% respondieron que si lo cual indica que la tasa positiva es baja, lo cua pude indicar el bajo interes o satisfacciòn del consumidor.

## Paso 12: Análisis de Quejas Abiertas por Coverage
1. Agrupa los datos por `Coverage` y calcula el número promedio de quejas abiertas (`Number of Open Complaints`).

Pregunta: ¿Existen coberturas con un número de quejas significativamente mayor? ¿Qué acciones se podrían tomar?

In [ ]:
df.groupby("Coverage")["Number of Open Complaints"].mean()

Coverage
Basic       0.384698
Extended    0.398979
Premium     0.333738
Name: Number of Open Complaints, dtype: float64

rta: si la cobertura extended tiene una mayor cantidad de quejas, lo que puede indicar que estos esten menos satisfechos con el producto. las acciones a seguir es primero analzar el por que es la categoria con más quejas, en segundo lugar se debe mejorar la comunicación con el cliente y se debe optimizar los proesos de atenición.

## Paso 13: Análisis de la Antigüedad de las Pólizas según EmploymentStatus
1. Agrupa por `EmploymentStatus` y calcula el promedio de `Months Since Policy Inception`.

Pregunta: ¿Existe alguna relación entre el estado laboral y la antigüedad de las pólizas? ¿Qué podría implicar esto en la estabilidad del cliente?

In [ ]:
df.groupby("EmploymentStatus")["Months Since Policy Inception"].mean()

EmploymentStatus
Disabled         47.469136
Employed         47.800456
Medical Leave    48.141204
Retired          48.531915
Unemployed       48.747087
Name: Months Since Policy Inception, dtype: float64

rta: si se puede evidenciar una relación emtre el estado labroral y la antiguedad de polizas puesto que las personas desempleadas, jubiladas y en licencia medica tieenen a tener polizas màs antiguas.

en la estabilidad del cliente esto puede implicar que los clientes con menos cambios en su vida laboral son aquellos que tienden a tener polizas màs estables , mientras que los clientes con una vida laboral màs activa pueden requerir incentivos para seguir en la aseguradora.


## Paso 14: Análisis del Total Claim Amount según Policy Type
1. Agrupa por `Policy Type` y calcula el Total Claim Amount promedio para cada tipo de póliza.

Pregunta: ¿Qué diferencias se observan entre los tipos de póliza y qué podrían implicar en términos de riesgo y eficiencia operativa?

In [ ]:
df.groupby("Policy Type")["Total Claim Amount"].mean()

Policy Type
Corporate Auto    430.312188
Personal Auto     434.828219
Special Auto      440.472770
Name: Total Claim Amount, dtype: float64

rta: las ddiferencias que se observan en los tipos de polizas son que el Total Claim Amount promedio es más alto en Special Auto $440.47, seguido de  Personal Auto $434.83 y finalmente corporate Auto $430.31.


esto puede implcar que eiste un mayor riesgo de polizas en special auto o que existe una diferencia en la eficiencia de estrategia operativa.

## Caso de Negocio: Evaluación de la Inversión en Marketing

### Contexto:
Una compañía de seguros ha asignado un presupuesto ficticio de **$5,000,000** para campañas de adquisición. El objetivo es evaluar la rentabilidad de la inversión comparando el gasto de marketing ( 50000000) (CAC) con el Customer Lifetime Value (CLV) obtenido.

### Tareas:
1. Utiliza el CLV promedio obtenido en el Paso 2.
2. Calcula el CAC promedio dividiendo el presupuesto de marketing entre el número total de clientes.
3. Compara el CAC promedio con el CLV promedio y responde:
   - ¿El CAC es inferior al CLV, lo que justificaría la inversión?
   - ¿Qué porcentaje del CLV representa el CAC?

_Reflexión estratégica: Si el CAC es significativamente menor que el CLV, la inversión en marketing es rentable. De lo contrario, se deberá replantear la estrategia de adquisición._

In [ ]:
marketing_budget = 5000000

In [ ]:
clv_promedio = df["Customer Lifetime Value"].mean()
print(f"Customer Lifetime Value (CLV) promedio: ${clv_promedio:.2f}")

Customer Lifetime Value (CLV) promedio: $8004.94


In [ ]:
total_customers = df.shape[0]
print(f"Costo de Adquisición de Clientes (CAC) promedio: ${cac_promedio:.2f}")

Costo de Adquisición de Clientes (CAC) promedio: $547.41


In [ ]:
cac_promedio = marketing_budget / total_customers
print(f"El CAC representa el {porcentaje_cac_sobre_clv:.2f}% del CLV.")

El CAC representa el 6.84% del CLV.


In [ ]:
rentabilidad = clv_promedio > cac_promedio
porcentaje_cac_sobre_clv = (cac_promedio / clv_promedio) * 100
if rentabilidad:
    print("✅ La inversión en marketing es rentable, ya que el CAC es menor que el CLV.")
else:
    print("❌ La inversión en marketing NO es rentable, ya que el CAC es mayor que el CLV. Se debe replantear la estrategia.")

✅ La inversión en marketing es rentable, ya que el CAC es menor que el CLV.


### **Bonus:** Teniendo lo anterior en cual es la proporcion de clientes rentables

In [ ]:
clientes_rentables = df[df["Customer Lifetime Value"] > cac_promedio].shape[0]
proporcion_rentables = (clientes_rentables / total_customers) * 100
print(f"Número de clientes rentables: {clientes_rentables}")
print(f"Proporción de clientes rentables: {proporcion_rentables:.2f}%")

Número de clientes rentables: 9134
Proporción de clientes rentables: 100.00%


## Conclusiones y Recomendaciones

### **Bonus 2:** En no más de un parrafo mencione 3 acciones concretas para mejorar los reusltados de la empresa.
